packages/eslint-plugin-stories/src/rules/no-new-components-without-story.ts

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+import type { Rule } from 'eslint';
+
+const failureMessage = `
+  React Components must have a corresponding Storybook story.
+
+  "Storybook" is our component explorer, and a "story" is a component example that serves as a unit for Storybook
+
+  Having a Storybook story for each component allows us to test our components' behavior, accessibility,
+  visual regression, etc. Learn more here:
+  https://go.czi.team/storybook-testing
+`;
+
+/**
+ * To encourage Storybook coverage, we warn developers about React components without stories.
+ */
+const rule: Rule.RuleModule = {
+  create(context) {
+    return {
+      ExportDefaultDeclaration(node) {
+        const filename = context.getFilename();
+        const isComponent =
+          filename.match('.tsx') &&
+          !filename.match('.stories.') &&
+          !filename.match('.spec.');
+
+        if (!isComponent) {
+          return;
+        }
+
+        const storyFilename = `${filename.split('.')[0]}.stories.tsx`;
+        const correspondingStoryExists = fs.existsSync(storyFilename);
+
+        if (!correspondingStoryExists) {
+          context.report({
+            node: node,
+            message: failureMessage,
+          });
+        }
+      },
+    };
+  },
+};
+
+export default rule;
```

Comment:
Nice job on the rule. You'll also need to import it this rule into index.ts and add it to the list of rules.

See https://github.com/chanzuckerberg/frontend-libs/blob/cfdf83f2d7efc568f0237a2ef632ac97bbe35d97/packages/eslint-plugin-stories/src/index.ts#L6

May also want to add it to the list of "strict" rules in the same file (but not recommended).

Reply:
Oh heh, that is ambiguous. I'm recommending:
- NOT adding the rule to `recommended`
- DO add the rule to `strict`